### PR TITLE
Rover: ouput range improvements

### PR DESCRIPTION
**Pull request by khancyr**
Hello,
This PR allow to use full power on on spot turn in skid steer config in manual mode. And limit SITL model input for motor between 1000 and 2000

### CODE REVIEW

File: libraries/AP_HAL_SITL/SITL_State.cpp [hunk 1]
```diff
@@ -385,9 +385,9 @@ void SITL_State::_simulator_servos(SITL::Aircraft::sitl_input &input)
         // add in engine multiplier
         if (input.servos[2] != 1500) {
             input.servos[2] = ((input.servos[2]-1500) * engine_mul) + 1500;
-            if (input.servos[2] > 2000) input.servos[2] = 2000;
-            if (input.servos[2] < 1000) input.servos[2] = 1000;
+            input.servos[2] = constrain_value(input.servos[2], 1000, 2000);
```
Comment (OXINARF) [inline libraries/AP_HAL_SITL/SITL_State.cpp]: You need to include AP_Math.h to use this.

**Comment by khancyr** [comment]
@OXINARF well bad idea to use constrain_value directly... so I revert to constrain_int16 (with a cast not mandatory but it shutdown warning about conversion from uint16_t to int16_t , and add no weight to binary)

File: APMrover2/Steering.cpp [hunk 2]
```diff
@@ -311,7 +311,11 @@ void Rover::set_servos(void) {
         if (!apply_skid_mix) {  // Mixage is already done by a controller so just pass the value to motor
             motor1 = steering_scaled;
             motor2 = throttle_scaled;
+        } else if (channel_throttle->in_trim_dz()) {  // Use full range for on spot turn
```
Comment (khancyr) [inline APMrover2/Steering.cpp]: well, this is blocking in all auto mode ... cause channel_throttle->in_trim_dz() is input from radio ...

**Comment by khancyr** [comment]
Ok! corrected and rebased

**Comment by khancyr** [comment]
rebased and correct regression from https://github.com/ArduPilot/ardupilot/commit/3236090d71373fd4eaab7239524523b28d48a83f

**Pull request merged**